### PR TITLE
Report v4 templates

### PR DESCRIPTION
**Pull request by lucernae**
Just to include the templates (jinja2 files) and Pep8 fix.
You don't have to review .html files, it will be changed again.
For less headache, just review .py files.

### CODE REVIEW

File: safe/reportv4/extractors/action_notes.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+def action_checklist_extractor(impact_report, component_metadata):
+    """
+    Extracting action checklist of the exposure layer
```
Comment (Gustry) [inline safe/reportv4/extractors/action_notes.py]: I think it should be on the same line as `"""` and finishing with a dot.

Comment (Gustry) [inline safe/reportv4/extractors/action_notes.py]: I think it should be on the same line as `"""` and finishing with a dot.

File: safe/reportv4/extractors/action_notes.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    :param impact_report: the impact report that acts as a proxy to fetch
+        all the data that extractor needed
+    :type impact_report: safe.reportv4.impact_report.ImpactReport
+    :param component_metadata: the component metadata. Used to obtain
```
Comment (Gustry) [inline safe/reportv4/extractors/action_notes.py]: Add space between parameter

File: safe/reportv4/extractors/action_notes.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    :type component_metadata: safe.reportv4.report_metadata.ReportMetadata
+
+    :return: context for rendering phase
+    :rtype: dict
+    """
+    context = {}
+
+    # figure out analysis report type
+    impact_layer = impact_report.impact_layer
+    """:type: qgis.core.QgsVectorLayer"""
```
Comment (Gustry) [inline safe/reportv4/extractors/action_notes.py]: What is this docstring ?

File: safe/reportv4/extractors/action_notes.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    """
+    Extracting notes and assumptions of the exposure layer
+
+    :param impact_report: the impact report that acts as a proxy to fetch
+        all the data that extractor needed
+    :type impact_report: safe.reportv4.impact_report.ImpactReport
+    :param component_metadata: the component metadata. Used to obtain
+        information about the component we want to render
+    :type component_metadata: safe.reportv4.report_metadata.ReportMetadata
+
```
Comment (Gustry) [inline safe/reportv4/extractors/action_notes.py]: Same comments as above for docstring (space and first line)

File: safe/reportv4/extractors/action_notes.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    :return: context for rendering phase
+    :rtype: dict
+    """
+    context = {}
+
+    # figure out analysis report type
+    impact_layer = impact_report.impact_layer
+    """:type: qgis.core.QgsVectorLayer"""
```
Comment (Gustry) [inline safe/reportv4/extractors/action_notes.py]: What is this docstring in the middle ? :)

File: safe/reportv4/extractors/analysis_result.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    classes = hazard_classification.get('classes')
+    result = []
+    for c in classes:
+        result.append({
+            'key': c.get('key'),
+            'name': c.get('name'),
+            'value': 0
+        })
+
+    hazard_class_idx = layer.fieldNameIndex('hazard_class')
```
Comment (Gustry) [inline safe/reportv4/extractors/analysis_result.py]: You should read the name field from the definitions.

File: safe/reportv4/extractors/analysis_result.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    hazard_class = hazard_classification.get('classes')
+    affected_dict = {}
+    for c in hazard_class:
+        affected_dict[c.get('key')] = c.get('affected')
+    result = {
+        'affected': 0,
+        'unaffected': 0,
+        'total': 0,
+    }
+    hazard_class_idx = layer.fieldNameIndex('hazard_class')
```
Comment (Gustry) [inline safe/reportv4/extractors/analysis_result.py]: Same the name field, we shouldn't hardcode these name field which are defined in definitions.

File: safe/reportv4/extractors/analysis_result.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
```
Comment (Gustry) [inline safe/reportv4/extractors/analysis_result.py]: Same thing here

File: safe/reportv4/extractors/action_notes.py (hunk 1 above)
Comment (lucernae) [inline safe/reportv4/extractors/action_notes.py]: Ah I forgot the convention.
Although, I really don't like it when it has to be on the same line as """. It looks ugly.

Comment (lucernae) [inline safe/reportv4/extractors/action_notes.py]: Same comment as above. I will fix it.

File: safe/reportv4/extractors/action_notes.py (hunk 2 above)
Comment (lucernae) [inline safe/reportv4/extractors/action_notes.py]: Docstring etc will be fixed at another PR

File: safe/reportv4/extractors/action_notes.py (hunk 3 above)
Comment (lucernae) [inline safe/reportv4/extractors/action_notes.py]: I used this for pycharm type hinting, really useful feature.
You will find some of these. Originally, I put this one since I code it backwards when 
impact_report still empty. But I deleted this one when the type in impact_report 
is fixed.
I'll delete any other if it is redundant with property type hinting and keep the rest where 
it is useful (such as variable in a loop).

File: safe/reportv4/extractors/action_notes.py (hunk 4 above)
Comment (lucernae) [inline safe/reportv4/extractors/action_notes.py]: Ok

File: safe/reportv4/extractors/analysis_result.py (hunk 7 above)
Comment (lucernae) [inline safe/reportv4/extractors/analysis_result.py]: Same as above.

File: safe/reportv4/extractors/analysis_result.py (hunk 8 above)
Comment (lucernae) [inline safe/reportv4/extractors/analysis_result.py]: Idem

**Comment by lucernae** [comment]
So I PR this since we want to fix that PEP8 things.
Review about docstrings and codes for the reporting it self will be fixed in this PR #3420 (Some of the diffs, were outdated too).
I Imagine #3420 will be long reviewed. So, I suggest we merge this if the PEP8 fix diffs is okay and everyone LGTM.

**Comment by Gustry** [comment]
As you said this PR should be merged, let's merge this. Comments will be fixed in the next PR as you said.
